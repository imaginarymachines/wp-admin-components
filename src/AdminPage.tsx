import React, { FC } from 'react';
const AdminPage: FC<{ children: any }> = ({ children }) => {
  return (
    <div className="wp-admin wp-core-ui js index-php auto-fold admin-bar customize-support svg sticky-menu">
      <div id="wpcontent">
        <div id="wpbody" role="main">
          <div id="wpbody-content">{children}</div>
        </div>
      </div>
    </div>
  );
};
export default AdminPage;
