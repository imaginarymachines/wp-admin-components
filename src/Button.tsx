import React, { FC, HTMLAttributes, ReactChild } from 'react';

export interface ButtonProps extends HTMLAttributes<HTMLDivElement> {
  /** custom content, defaults to 'the snozzberries taste like snozzberries' */
  children?: ReactChild;
  variant?: 'primary' | 'secondary';
  a?:"string";
  onClick?: () => void;
  isSmall?:boolean;
  isHero?:boolean;
}

const Button: FC<ButtonProps> = ({ children,variant,a,onClick,isSmall,isHero }) => {
   let className = `button  button-${variant ? variant : 'primary'}`;
   if( isSmall ){
     className = `${className} button-small`;
   }
   if( isHero ){
    className = `${className} button-hero`;
  }
   if( a ){
    return <a href={a} className={className}>{children}</a>;

   }
   return <button onClick={onClick} className={className}>{children}</button>
};
export default Button;
