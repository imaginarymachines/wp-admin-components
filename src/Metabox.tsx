import React, { FC,ReactChild } from 'react';

export interface MetaboxProps { children: ReactChild,title:string };
const Metabox: FC<MetaboxProps> = ({ children,title }) => {
  return (

    <div id="dashboard-widgets" className="metabox-holder">
    <div id="postbox-container-1" className="postbox-container">
        <div id="normal-sortables" className="meta-box-sortables ui-sortable">
            <div id="metabox" className="postbox">
                <button type="button" className="handlediv" aria-expanded="true">
                    <span className="screen-reader-text">Metabox collapse</span>
                    <span className="toggle-indicator" aria-hidden="true"></span>
                </button>
                <h2 className="hndle ui-sortable-handle">
                    <span>
                        {title}
                    </span>
                </h2>
                <div className="inside">
                    <div className="main">
                        {children}
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>

  );
};
export default Metabox;
