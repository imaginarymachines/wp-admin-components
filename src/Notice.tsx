import React, {FC, useMemo}from'react';

export type NoticeProps = {
    heading?:string;
    link?:string;
    description:string;
    type?:'success'|'error'|'info'|'warning',
    onDismiss?:() => void;
}
export const Notice: FC<NoticeProps>= ({ heading, link, description, type,onDismiss }) => {
    const className = useMemo( () => {
        let cN = `notice ${type ?`notice-${type}`:''}`;
        if( onDismiss){
            cN = `${cN} is-dismissible`;
        }
        return cN;
    },[type,onDismiss]);
    return(
    <div className={className}>
      {heading? <h2>{heading}</h2>:null}
      <p className="description">
          {link ? (
             <a href={link} rel="noopener" target="_blank">
               {description}
             </a>
          ): (
            <>{description}   </>
          )
        }
      </p>
    </div>
  );
}
