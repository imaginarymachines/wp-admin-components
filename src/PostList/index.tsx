import React, {FC,ReactChild} from 'react';


export type PostListItemProps = {children:ReactChild};
export const PostListItem : FC<PostListItemProps> = ({children}) => {

    return (<>{children}</>);
}

export type PostListProps = {children:ReactChild};

export const PostList : FC<{children:PostListProps}> = ({children}) => {
    return (
        <table className="wp-list-table widefat fixed striped posts">
            <thead>
                <tr>
                    <td id="cb" className="manage-column column-cb check-column">
                        <label className="screen-reader-text" htmlFor="cb-select-all-1">Select All</label>
                        <input id="cb-select-all-1" type="checkbox" />
                    </td>
                    <th scope="col" id="title" className="manage-column column-title column-primary sortable desc">
                        <span>Title</span>
                    </th>
                    <th scope="col" id="author" className="manage-column column-author">Author</th>
                    <th scope="col" id="categories" className="manage-column column-categories">Categories</th>
                    <th scope="col" id="tags" className="manage-column column-tags">Tags</th>
                    <th scope="col" id="comments" className="manage-column column-comments num sortable desc">
                        <span>
                            <span className="vers comment-grey-bubble" title="Comments">
                                <span className="screen-reader-text">Comments</span>
                            </span>
                        </span>
                    </th>
                    <th scope="col" id="date" className="manage-column column-date sortable asc">
                        <a href="#">
                            <span>Date</span><span className="sorting-indicator"></span>
                        </a>
                    </th>
                </tr>
            </thead>
            <tbody id="the-list">
                {children}
            </tbody>
        </table>
    )
}
