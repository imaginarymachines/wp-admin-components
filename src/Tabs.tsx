import React, {
  FC,
  HTMLAttributes,
  ReactChild,
  useMemo,
  useState,
} from 'react';

export interface TabProps {
  id: string;
  children: ReactChild;
  label: string;
}

export const Tab: FC<TabProps> = ({ id, children }) => {
  return <div id={id}>{children}</div>;
};

export interface TabButtonProps {
  id: string;
  label: ReactChild;
  isSelected: boolean;
  onClick: (id: string) => void;
}

export const TabButton: FC<TabButtonProps> = ({
  id,
  label,
  isSelected,
  onClick,
}) => {
  return (
    <li
      id={`tab-button-${id}`}
      aria-controls={`tab-${id}`}
      aria-selected={isSelected}
    >
      <a
        className="nav-tab"
        href={`#${id}`}
        onClick={(e) => {
          e.preventDefault();
          onClick(id);
        }}
      >
        {label}
      </a>
    </li>
  );
};

export interface TabContentProps extends HTMLAttributes<HTMLDivElement> {
  id: string;
  children: ReactChild;
  isSelected: boolean;
}

export const TabContent: FC<TabContentProps> = ({
  id,
  children,
  isSelected,
}) => {
  return (
    <div
      id={`tab-${id}`}
      className="tabs-content"
      role="tabpanel"
      aria-labelledby={`tab-button-${id}`}
      aria-hidden={!isSelected}
      style={
        isSelected
          ? {
              display: 'block',
              visibility: 'visible',
            }
          : {
              display: 'none',
              visibility: 'hidden',
            }
      }
    >
      {children}
    </div>
  );
};

export interface TabsProps extends HTMLAttributes<HTMLDivElement> {
  id: string;
  tabs: TabProps[];
  initialTab?: string;
}

export const Tabs: FC<TabsProps> = (props) => {
  const tabs = useMemo(() => {
    return props.tabs.length ? props.tabs : [];
  }, [props.tabs]);
  const [selectedTab, setSelectedTab] = useState<string | null>(() => {
    if (props.initialTab) {
      return props.initialTab;
    }
    if (!tabs.length) {
      return null;
    }
    return tabs[0].id;
  });

  const isSelectedTab = (tab: TabProps) => selectedTab === tab.id;

  return (
    <div id={props.id}>
      <h2 className="nav-tab-wrapper">
        <ul role="tablist">
          {tabs.map((tab: TabProps) => (
            <TabButton
              key={tab.id}
              isSelected={isSelectedTab(tab)}
              id={tab.id}
              label={tab.label}
              onClick={setSelectedTab}
            />
          ))}
        </ul>
      </h2>

      <>
        {tabs.map((tab: TabProps) => (
          <TabContent key={tab.id} isSelected={isSelectedTab(tab)} id={tab.id}>
            {tab.children}
          </TabContent>
        ))}
      </>
    </div>
  );
};
