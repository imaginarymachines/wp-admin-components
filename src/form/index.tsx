export { Form, FormProps, FormTableProps, FormTable } from './FormTable';
export { FieldTr } from './FieldTr';
export { TrInput, TrSelect, TrSubmitButton } from './FormTableFields';
