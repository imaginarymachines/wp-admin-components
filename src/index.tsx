import Button, { ButtonProps } from './Button';
import { Tabs, TabsProps, TabProps } from './Tabs';
import { Notice, NoticeProps } from './Notice';
import Metabox, {
  MetaboxProps,MetaboxWrapper,
  MetaboxWrapperProps, } from './Metabox';
import {
  FormTable,
  FormTableProps,
  Form,
  FormProps,
  FieldTr,
  TrInput,
  TrSelect,
  TrSubmitButton,
} from './form';
export {
  Button,
  ButtonProps,
  Tabs,
  TabsProps,
  TabProps,
  FormTable,
  FormTableProps,
  FormProps,
  Form,
  FieldTr,
  TrInput,
  TrSelect,
  TrSubmitButton,
  Notice,
  NoticeProps,
  Metabox,
  MetaboxProps,
  MetaboxWrapper,
  MetaboxWrapperProps,
};
