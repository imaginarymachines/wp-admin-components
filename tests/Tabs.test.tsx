import {Tabs} from '../src';
import React from 'react';
import snapshot from './snapshot';
const tabs = [
    {id: 'one', children:<div>Tab One Content</div>,label:'One'},
    {id: 'two', children:<div>Tab Two Content</div>,label:'Two'},
    {id: 'three', children:<div>Tab Three Content</div>,label:'Three'},
];

describe( 'Tabs',() => {
    it( 'Renders', () => {
        snapshot(<Tabs id="tabs-1" tabs={tabs} />)
    });
    it( 'Renders with inital tab', () => {
        snapshot(<Tabs id="tabs-2" tabs={tabs}  initialTab={'two'}/>)
    });

});
