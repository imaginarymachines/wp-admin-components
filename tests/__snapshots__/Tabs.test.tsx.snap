// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Renders 1`] = `
<div>
  <div
    id="tabs-1"
  >
    <h2
      class="nav-tab-wrapper"
    >
      <ul
        role="tablist"
      >
        <li
          aria-controls="tab-one"
          aria-selected="true"
          id="tab-button-one"
        >
          <a
            class="nav-tab"
            href="#one"
          >
            One
          </a>
        </li>
        <li
          aria-controls="tab-two"
          aria-selected="false"
          id="tab-button-two"
        >
          <a
            class="nav-tab"
            href="#two"
          >
            Two
          </a>
        </li>
        <li
          aria-controls="tab-three"
          aria-selected="false"
          id="tab-button-three"
        >
          <a
            class="nav-tab"
            href="#three"
          >
            Three
          </a>
        </li>
      </ul>
    </h2>
    <div
      aria-hidden="false"
      aria-labelledby="tab-button-one"
      class="tabs-content"
      id="tab-one"
      role="tabpanel"
      style="display: block; visibility: visible;"
    >
      <div>
        Tab One Content
      </div>
    </div>
    <div
      aria-hidden="true"
      aria-labelledby="tab-button-two"
      class="tabs-content"
      id="tab-two"
      role="tabpanel"
      style="display: none; visibility: hidden;"
    >
      <div>
        Tab Two Content
      </div>
    </div>
    <div
      aria-hidden="true"
      aria-labelledby="tab-button-three"
      class="tabs-content"
      id="tab-three"
      role="tabpanel"
      style="display: none; visibility: hidden;"
    >
      <div>
        Tab Three Content
      </div>
    </div>
  </div>
</div>
`;

exports[`Tabs Renders with inital tab 1`] = `
<div>
  <div
    id="tabs-2"
  >
    <h2
      class="nav-tab-wrapper"
    >
      <ul
        role="tablist"
      >
        <li
          aria-controls="tab-one"
          aria-selected="false"
          id="tab-button-one"
        >
          <a
            class="nav-tab"
            href="#one"
          >
            One
          </a>
        </li>
        <li
          aria-controls="tab-two"
          aria-selected="true"
          id="tab-button-two"
        >
          <a
            class="nav-tab"
            href="#two"
          >
            Two
          </a>
        </li>
        <li
          aria-controls="tab-three"
          aria-selected="false"
          id="tab-button-three"
        >
          <a
            class="nav-tab"
            href="#three"
          >
            Three
          </a>
        </li>
      </ul>
    </h2>
    <div
      aria-hidden="true"
      aria-labelledby="tab-button-one"
      class="tabs-content"
      id="tab-one"
      role="tabpanel"
      style="display: none; visibility: hidden;"
    >
      <div>
        Tab One Content
      </div>
    </div>
    <div
      aria-hidden="false"
      aria-labelledby="tab-button-two"
      class="tabs-content"
      id="tab-two"
      role="tabpanel"
      style="display: block; visibility: visible;"
    >
      <div>
        Tab Two Content
      </div>
    </div>
    <div
      aria-hidden="true"
      aria-labelledby="tab-button-three"
      class="tabs-content"
      id="tab-three"
      role="tabpanel"
      style="display: none; visibility: hidden;"
    >
      <div>
        Tab Three Content
      </div>
    </div>
  </div>
</div>
`;
